src/zmq/zmqpublishnotifier.cpp

```diff
@@ -178,6 +180,16 @@ bool CZMQPublishHashTransactionNotifier::NotifyTransaction(const CTransaction &t
     return SendMessage(MSG_HASHTX, data, 32);
 }
 
+bool CZMQPublishHashTransactionEvictionNotifier::NotifyTransactionEviction(const CTransaction &transaction, MemPoolRemovalReason &reason)
+{
+    uint256 hash = transaction.GetHash();
+    LogPrint(BCLog::ZMQ, "zmq: Publish hashtxevict %s\n", hash.GetHex());
+    char data[32];
+    for (unsigned int i = 0; i < 32; i++)
```

Comment:
Do we want to report the reason here? No strong reason here. On one hand, reasons are an  implementation detail, on the other, for statistics it could be useful to report it (I think back in the day when I had a patch to add this functionality, I did add the reason, though converted to a interface-specific enum).

Reply:
Just a hunch but I think in general "no". Anyone who wants to do statistics on mempool drops can look at logs after the fact? The aim of this PR is primarily for timely notification of transaction changes in "known" mempool.